File: JAVA/2021/1stOCT_abhinandan-khurana.java
import java.util.*;
public class Main
{
    static boolean prime (int a)
    {
        for(int i=2;i<=(int)Math.sqrt(a);i++)
        {
            if(a%i==0)
            return false;
        }
        return true;
    }
	public static void main(String[] args) {
		  int n1=1,n2=1,n3,i,count;
		  Scanner sc=new Scanner(System.in);
		  count = sc.nextInt();
          System.out.print(n1+" "+n2);       
    
          for(i=2;i<count;++i)        
            {    
                n3=n1+n2;
                if(prime(n3) || n3%5==0)
                System.out.print(" 0");
                else
                System.out.print(" "+n3);    
                n1=n2;    
                n2=n3;    
	       }
	}    
}

File: Python/2021/1stOCT_abhinandan-khurana.py
from math import sqrt
nterms = int(input())
n1, n2 = 1, 1
count = 0


lst= []
while count < nterms:
  lst.append(n1)
  nth = n1 + n2
  n1 = n2
  n2 = nth
  count += 1


def prime(a):
  flag = True
  if(a > 1):
    for i in range(2, int(sqrt(a)) + 1):
        if (a % i == 0):
            flag = False
            break
    if(a % 5 == 0):
      flag = True
  return flag

k = set(lst)
k = list(lst)
for i in lst:
  if(prime(i)):
    k.remove(i)


for i in range(2,len(lst)):
  if(lst[i] in k):
    continue
  else:
    lst[i] = 0
print(lst)
